hal/network/ncp/wifi/wlan_hal.cpp

```diff
@@ -428,36 +428,21 @@ int softap_set_application_page_handler(PageProvider* provider, void* reserved)
     return SYSTEM_ERROR_NOT_SUPPORTED;
 }
 
-static bool isValidCountryCode(const char * country_code) {
-    if(strlen(country_code) != 2){
-        return false;
-    }
-
-    String country_string(country_code);
-
-    if (country_string == "US" || country_string == "CA") {
-        return true;
-    } else {
-        return false;
-    }
+int wlan_set_country_code(wlan_country_code_t country_code, const void* reserved) {
+    char cc_buf[2] = { (char)(country_code >> 8),  (char)(country_code & 0xFF) };
+    return dct_write_app_data(cc_buf, DCT_COUNTRY_CODE_OFFSET, 2);
 }
 
-int wlan_set_country_code(const char * country_code) {
-    if (isValidCountryCode(country_code)) {
-        return dct_write_app_data(country_code, DCT_COUNTRY_CODE_OFFSET, 2);
-    } else {
-        return SYSTEM_ERROR_NOT_SUPPORTED;
+wlan_country_code_t wlan_get_country_code(void* reserved) {
+     char cc_buf[2] = {};
+    wlan_country_code_t country_code = wlan_country_code_t::WLAN_CC_UNSET;
+    int rc = dct_read_app_data_copy(DCT_COUNTRY_CODE_OFFSET, &cc_buf, 2);
+    if (0 == rc) {
```

Comment:
Nit: we don't normally use this pattern, let's avoid introducing it, as it decreases readability. We all know why this pattern was used in C/C++ (to avoid using assignment operator vs comparison). This is caught by warnings in our codebase, no need for it.